### DISCUSSION TITLE
v4: issues with multilingual site

### DISCUSSION THREAD

**Discussion by Magneticdud** [Lednerb/bilberry-hugo-theme#626] [Q&A] [answered]
Hello, I have two websites, example.com (english) and example.it (italian)

One year ago, I set it correctly, it was generating ok: two separate directories that I would upload to the two different FTPs

now that i changed to v4 in hugo.toml, I have two issues:

1. in the example.it italian website, theme.css and theme.js are loaded from example.com, and it gives a CORS error in console. The files are not loaded by the browser and so the page is visually broken. I set cloudflare to add Access-Control-Allow-Origin=* to all URLs and I can pass [this test](https://cors-test.codehappy.dev/) but nothing changes

2. for the secondary language, the resized version of featuredImage.jpg is not generated, so it gives a 404 and the article is without the image

3. the bottom "language change" button correctly links to the right URL, but the link at the top of the page, is to self at both languages. This works fine if baseURL is not set. Once i set, only the bottom links are correct

what mistake in the configuration could I've done?

edit:

for the issue of the images I see this when i generate:

```
Start building sites … 
hugo v0.123.1-3f8434a62fc91a50f4e5bd43a4435d2d301844ff+extended linux/amd64 BuildDate=2024-02-21T08:17:45Z VendorInfo=gohugoio


                   | EN | IT  
-------------------+----+-----
  Pages            | 42 | 42  
  Paginator pages  |  0 |  0  
  Non-page files   | 77 | 77  
  Static files     | 91 | 91  
  Processed images |  1 |  0  
  Aliases          | 11 | 11  
  Cleaned          |  0 |  0
```

so it's not processing that image, for some reason...

**Comment by igor-baiborodine** [comment]
@Magneticdud, do I understand correctly that, technically speaking, it's not an issue of a multilingual site since you deploy two different websites? If that's the case, you should have two different `config.toml` files and build them separately. Then, please provide me with a link to the GitHub repository where I can examine the content of your websites and the commands you use to build them.

**Reply by Magneticdud** [reply]
no, it's a single website that's generating two directories in the output, /en and /it

I didn't use a github repo so i recreated it here

https://github.com/Magneticdud/Forexperiments-temp

URLs are: https://www.forexperimenti.it/posts/2024-03-03-finally-migrated/ and https://www.forexperiments.com/posts/2024-03-03-finally-migrated/

note if you see weird redirections is that the server redirects 404s to the old wordpress domain, so for example the missing resized featuredImage.jpg for the italian version

the CORS should have been set correctly, test result: https://cors-test.codehappy.dev/?url=https%3A%2F%2Fwww.forexperiments.com&origin=https%3A%2F%2Fwww.forexperimenti.it&method=get

**Comment by Magneticdud** [comment]
I found a solution for the CORS problem. After I edited the css and js partials, i visually fixed my website. I go to eat then submit a PR for this

**Reply by Magneticdud** [reply]
the PR: https://github.com/Lednerb/bilberry-hugo-theme/pull/628

**Comment by Magneticdud** [comment]
regarding the problem of "language change", `layouts/partials/article-footer.html` works fine, produces the correct links when using separate domains, while `layouts/partials/header.html` and `layouts/partials/footer.html` only work correctly when using directories for language change. I will investigate this in the next days

**Reply by Magneticdud** [reply, marked as answer]
a small fix for this is in PR #694
